cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -923,6 +927,66 @@ public KafkaBrokerConfigurationBuilder withTieredStorage(String clusterName, Tie
         return this;
     }
 
+    /**
+     * Configures the quotas based on the type of the plugin - {@link QuotasPluginKafka}, {@link QuotasPluginStrimzi}
+     *
+     * @param clusterName    Name of the cluster
+     * @param quotasPlugin   Configuration of the quotas plugin
+     * @return  Returns the builder instance
+     */
+    public KafkaBrokerConfigurationBuilder withQuotas(String clusterName, QuotasPlugin quotasPlugin) {
+        if (quotasPlugin != null) {
+            // for the built-in Kafka quotas plugin we don't need to configure anything
+            if (quotasPlugin instanceof QuotasPluginStrimzi quotasPluginStrimzi) {
+                printSectionHeader("Quotas configuration");
+
+                configureQuotasPluginStrimzi(clusterName, quotasPluginStrimzi);
+
+                writer.println();
+            }
+        }
+
+        return this;
+    }
+
+    /**
+     * Configures Strimzi quotas plugin
+     *
+     * @param clusterName           Name of the cluster
+     * @param quotasPluginStrimzi   Strimzi quotas plugin configuration
+     */
+    private void configureQuotasPluginStrimzi(String clusterName, QuotasPluginStrimzi quotasPluginStrimzi) {
+        // add TopicOperator's and CruiseControl's user to the excluded principals
+        List<String> excludedPrincipals = new ArrayList<>(List.of(
+            String.format("User:CN=%s,O=io.strimzi", KafkaResources.kafkaComponentName(clusterName)),
+            String.format("User:CN=%s,O=io.strimzi", CruiseControlResources.componentName(clusterName))
+        ));
+
+        writer.println(String.format("%s=%s", "client.quota.callback.class", "io.strimzi.kafka.quotas.StaticQuotaCallback"));
+
+        // configuration of Admin client that will check the cluster
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.bootstrap.servers", clusterName + "-kafka-brokers:9091"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.security.protocol", "SSL"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.ssl.keystore.location", "/tmp/kafka/cluster.keystore.p12"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.ssl.keystore.password", "${CERTS_STORE_PASSWORD}"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.ssl.keystore.type", "PKCS12"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.ssl.truststore.location", "/tmp/kafka/cluster.truststore.p12"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.ssl.truststore.password", "${CERTS_STORE_PASSWORD}"));
+        writer.println(String.format("%s=%s", "client.quota.callback.static.kafka.admin.ssl.truststore.type", "PKCS12"));
```

Comment:
Maybe you should check the `withCruiseControl` method to see how this is done there?

Reply:
This is how it is done in the `configureOpaAuthorization`, from where I took the inspiration. If it's a problem, I can do it like in the `withCruiseControl`

Reply:
OPA authorizer doesn't connect to Kafka. So it isnt the right thing to _copy_,.